client/src/main/kotlin/io/spine/examples/pingh/client/MentionsFlow.kt

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2024, TeamDev. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.examples.pingh.client
+
+import com.google.protobuf.Duration
+import com.google.protobuf.util.Timestamps
+import io.spine.base.Time.currentTime
+import io.spine.examples.pingh.mentions.GitHubClientId
+import io.spine.examples.pingh.mentions.MentionId
+import io.spine.examples.pingh.mentions.MentionStatus
+import io.spine.examples.pingh.mentions.MentionView
+import io.spine.examples.pingh.mentions.UserMentions
+import io.spine.examples.pingh.mentions.UserMentionsId
+import io.spine.examples.pingh.mentions.buildBy
+import io.spine.examples.pingh.mentions.command.MarkMentionAsRead
+import io.spine.examples.pingh.mentions.command.SnoozeMention
+import io.spine.examples.pingh.mentions.command.UpdateMentionsFromGitHub
+import io.spine.examples.pingh.mentions.event.MentionRead
+import io.spine.examples.pingh.mentions.event.MentionSnoozed
+import io.spine.examples.pingh.mentions.event.MentionsUpdateFromGitHubCompleted
+import java.lang.Thread.sleep
+import kotlinx.coroutines.flow.MutableStateFlow
+
+/**
+ * The flow for managing the lifecycle of mentions.
```

Comment:
Let's rewrite:

Controls the lifecycle of mentions and handles the user's action in relation to them.

In particular:

* triggers the periodical update of the data via GitHub API,  (TODO: is it so? I am purely guessing — Alex.)
* whenever the user acts in relation to some mention, propagates these actions as commands to the server-side.